docs/csharp/codedoc.md

```diff
@@ -12,16 +12,464 @@ ms.devlang: dotnet
+* [&lt;c&gt;](#c)
+* [&lt;code&gt;](#code)
+* [&lt;para&gt;](#para)
+* [&lt;list&gt;](#list)
+* [&lt;see&gt;](#see)
+* [&lt;seealso&gt;](#seealso)
+* [&lt;param&gt;](#param)
+* [&lt;paramref&gt;](#paramref)
+* [&lt;typeparam&gt;](#typeparam)
+* [&lt;typeparamref&gt;](#typeparamref)
+* [&lt;include&gt;](#include)
+
+<a name="summary"></a>
+## &lt;summary&gt;
+
+The &lt;summary&gt; tag contains the primary information of a class or method.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <summary>
+        /// This method does stuff
+        /// </summary>
+        public void DoStuff(int a)
+        {
+
+        }
+
+        /// <summary>
+        /// This method does more stuff
+        /// </summary>
+        public void DoMoreStuff(int a, string b)
+        {
+
+        }
+    }
+
+<a name="remarks"></a>
+## &lt;remarks&gt;
+
+The &lt;remarks&gt; tag is used to add information about a type, supplementing the information specified with &lt;summary&gt;.
+
+    /// <summary>
+    /// This class primarily does something
+    /// </summary>
+    /// <remarks>
+    /// It can also do everything
+    /// </remarks>
+    public class SomeClass
+    {
+        
+    }
+
+<a name="returns"></a>
+## &lt;returns&gt;
+
+The &lt;returns&gt; tag should be used in the comment for a method declaration to describe the return value.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <returns>Stuff it has done</returns>
+        public int DoStuff(int a)
+        {
+            return a;
+        }
+    }
+
+<a name="example"></a>
+## &lt;example&gt;
+
+The &lt;example&gt; tag lets you specify an example of how to use a method or other library member. 
+This commonly involves using the [&lt;code&gt;](#code) tag.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <example>
+        /// <code>
+        /// SomeClass someClass = new SomeClass();
+        /// someClass.DoStuff();
+        /// </code>
+        /// </example>
+        public void DoStuff()
+        {
+
+        }
+    }
+
+<a name="value"></a>
+## &lt;value&gt;
+
+The &lt;value&gt; lets you describe the value that a property represents.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <value>The Name property gets/sets the name value of something</value>
+        public string Name { get; set; }
+    }
+
+<a name="exception"></a>
+## &lt;exception&gt;
+
+The &lt;exception&gt; tag lets you specify possible exceptions that can be thrown. 
+It's applicable to methods, properties, events, and indexers.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <exception cref="System.Exception">Thrown when there's no stuff to do</exception>
```

Comment:
I think it's important to show an example that has two different types of exceptions thrown:

e.g. an ArgumentNullException, and an InvalidOperationException.

Further, I'd avoid using System.Exception as the thrown type. That's not a practice we'd want to encourage. Handling that requires catching all exceptions, and that exception type carries very little information.